lib/pages/tabs/4_settings.dart

```diff
@@ -527,405 +501,280 @@ class _SettingsState extends State<Settings>
-                                                  ),
-                                                ),
-                                              ],
-                                              mainAxisAlignment:
-                                                  MainAxisAlignment
-                                                      .spaceBetween,
-                                            ),
-                                          ),
-                                          Expanded(
-                                            child: Directionality(
-                                              textDirection: TextDirection.ltr,
-                                              child: CupertinoTheme(
-                                                data: CupertinoThemeData(
-                                                  textTheme:
-                                                      CupertinoTextThemeData(
-                                                    dateTimePickerTextStyle:
-                                                        Theme.of(context)
-                                                            .textTheme
-                                                            .bodyText1,
-                                                  ),
-                                                ),
-                                                child: CupertinoDatePicker(
-                                                  backgroundColor:
-                                                      Theme.of(context)
-                                                          .cardColor,
-                                                  mode: CupertinoDatePickerMode
-                                                      .time,
-                                                  initialDateTime:
-                                                      morningNotiTime ??
-                                                          DateTime.now(),
-                                                  onDateTimeChanged:
-                                                      (DateTime newDateTime) {
-                                                    setState(() {
-                                                      morningTemp = newDateTime;
-                                                    });
-                                                  },
-                                                  minuteInterval: 1,
-                                                ),
-                                              ),
-                                            ),
-                                          ),
-                                        ],
+                  NotificationItem(
+                    text: 'التنبيه لأذكار الصباح',
+                    icon: NoorIcons.morning,
+                    time: formatTime(morningNotiTime),
+                    enabled: morningNotiEnabled,
+                    onTap: () {
```

Comment:
onTap method is redundant, it could be made into one method and called from morning & night widgets

Reply:
ok, what if we tried something like this:
```suggestion
                    onTap: () async {
                      final DateTime? selectedTime = await pickTime(context);
                      // ... 
                    }
                    
                    
Future<DateTime?> pickTime(BuildContext context) async {
    DateTime? timeChange;
    final DateTime? time = await showModalBottomSheet<DateTime?>(
      isDismissible: false,
      context: context,
      builder: (BuildContext context) => SizedBox(
        height: 250,
        child: Column(
          children: <Widget>[
            Container(
              height: 35,
              color: Theme.of(context).dialogTheme.backgroundColor,
              child: Row(
                children: <Widget>[
                  SizedBox(
                    width: 60,
                    child: TextButton(
                      child: Text(
                        'حفظ',
                        style: Theme.of(context).textTheme.bodyText1!.copyWith(
                            fontWeight: FontWeight.bold,
                            color: Theme.of(context).accentColor,
                            fontSize: 12),
                      ),
                      onPressed: () async {
                        Navigator.of(context).pop(timeChange);
                      },
                    ),
                  ),
                  SizedBox(
                    width: 60,
                    child: TextButton(
                      child: Text(
                        'إلغاء',
                        style: Theme.of(context).textTheme.bodyText1!.copyWith(
                            fontWeight: FontWeight.bold,
                            color: Theme.of(context).accentColor,
                            fontSize: 12),
                      ),
                      onPressed: () {
                        Navigator.of(context).pop();
                      },
                    ),
                  ),
                ],
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
              ),
            ),
            Expanded(
              child: Directionality(
                textDirection: TextDirection.ltr,
                child: CupertinoTheme(
                  data: CupertinoThemeData(
                    textTheme: CupertinoTextThemeData(
                      dateTimePickerTextStyle:
                          Theme.of(context).textTheme.bodyText1,
                    ),
                  ),
                  child: CupertinoDatePicker(
                    backgroundColor: Theme.of(context).cardColor,
                    mode: CupertinoDatePickerMode.time,
                    initialDateTime: morningNotiTime ?? DateTime.now(),
                    onDateTimeChanged: (DateTime newDateTime) =>
                        timeChange = newDateTime,
                    minuteInterval: 1,
                  ),
                ),
              ),
            ),
          ],
        ),
      ),
    );

    return time;
  }
```